### PR TITLE
Shuffle MC and SC questions

### PR DESCRIPTION
**Pull request by kjuej**

### CODE REVIEW

File: src/teaching-element/Assessment/MultipleChoice.vue [hunk 1]
```diff
@@ -36,22 +37,32 @@ export default {
...
       return options.type;
+    },
+    randomAnswers() {
+      return shuffle(this.answers);
     }
   },
   methods: {
     update() {
+      this.userAnswer = [];
+      this.unorderedAnswer.forEach(a => {
```
Comment (abasic) [inline src/teaching-element/Assessment/MultipleChoice.vue]: This logic can be a computed meaning in this assessment the userAnswer should be a computed.

File: src/teaching-element/Assessment/MultipleChoice.vue [hunk 2]
```diff
@@ -23,6 +23,7 @@
 <script>
 import includes from 'lodash/includes';
 import isEmpty from 'lodash/isEmpty';
+import shuffle from 'lodash/shuffle';
```
Comment (abasic) [inline src/teaching-element/Assessment/MultipleChoice.vue]: The order is not correct.

File: src/teaching-element/Assessment/MultipleChoice.vue [hunk 3]
```diff
@@ -36,22 +37,32 @@ export default {
...
     }
   },
   methods: {
     update() {
+      this.userAnswer = [];
+      this.unorderedAnswer.forEach(a => {
+        this.userAnswer.push(this.answers.indexOf(a));
+      });
       let { userAnswer } = this;
       userAnswer = isEmpty(userAnswer) ? null : userAnswer.sort();
```
Comment (abasic) [inline src/teaching-element/Assessment/MultipleChoice.vue]: No need for the `isEmpty` here.

File: src/teaching-element/Assessment/SingleChoice.vue [hunk 4]
```diff
@@ -33,20 +34,28 @@ export default {
...
       return options.type;
+    },
+    shuffledAnswers() {
+      return shuffle(this.answers);
     }
   },
   methods: {
     update() {
+      let { shuffledAnswers, unorderedAnswer, answers } = this;
+      this.userAnswer = answers.indexOf(shuffledAnswers[unorderedAnswer]);
```
Comment (abasic) [inline src/teaching-element/Assessment/SingleChoice.vue]: This can be a computed as well.

**Comment by zcuric** [comment]
@kronicker CR?

File: src/teaching-element/Assessment/MultipleChoice.vue [hunk 5]
```diff
@@ -36,25 +36,27 @@ export default {
     submission: { type: Array, default: () => ([]) }
   },
   data() {
-    return { userAnswer: this.submission || [] };
+    return {
+      userAnswer: this.submission || []
+    };
```
Comment (kronicker) [inline src/teaching-element/Assessment/MultipleChoice.vue]:
```js
data: () => ({ userAnswer: this.submission || [] })
```

File: src/teaching-element/Assessment/MultipleChoice.vue [hunk 6]
```diff
@@ -36,25 +36,27 @@ export default {
...
   },
   computed: {
-    type() {
-      const options = this.options.multipleChoice || defaults;
-      return options.type;
+    config: vm => ({ ...defaults, ...vm.options.multipleChoice }),
+    choices() {
+      const { isRandom } = this.config;
```
Comment (kronicker) [inline src/teaching-element/Assessment/MultipleChoice.vue]: `randomize` or something similar (it's not a state but a feature flag)

File: src/teaching-element/Assessment/SingleChoice.vue [hunk 7]
```diff
@@ -33,12 +35,16 @@ export default {
     submission: { type: Number, default: null }
   },
   data() {
-    return { userAnswer: this.submission };
+    return {
+      userAnswer: this.submission
+    };
```
Comment (kronicker) [inline src/teaching-element/Assessment/SingleChoice.vue]: As ⬆️

File: src/teaching-element/Assessment/SingleChoice.vue [hunk 8]
```diff
@@ -33,12 +35,16 @@ export default {
...
   },
   computed: {
-    type() {
-      const options = this.options.singleChoice || defaults;
-      return options.type;
+    config: vm => ({ ...defaults, ...vm.options.singleChoice }),
+    choices() {
+      const { isRandom } = this.config;
```
Comment (kronicker) [inline src/teaching-element/Assessment/SingleChoice.vue]: As ⬆️

**Pull request merged**